src/daft-table/src/ops/joins/mod.rs

```diff
@@ -103,14 +108,96 @@ pub fn infer_join_schema(
     Schema::new(join_fields)
 }
 
+fn add_non_join_key_columns(
+    left: &Table,
+    right: &Table,
+    lidx: Series,
+    ridx: Series,
+    left_on: &[ExprRef],
+    right_on: &[ExprRef],
+    mut join_series: Vec<Series>,
+) -> DaftResult<Vec<Series>> {
+    let mut names_so_far = join_series
+        .iter()
+        .map(|s| s.name().to_string())
+        .collect::<HashSet<_>>();
+
+    // TODO(Clark): Parallelize with rayon.
+    for field in left.schema.fields.values() {
+        if names_so_far.contains(&field.name) {
+            continue;
+        } else {
+            join_series.push(left.get_column(&field.name)?.take(&lidx)?);
+            names_so_far.insert(field.name.clone());
+        }
+    }
+
+    drop(lidx);
+
+    // Zip the names of the left and right expressions into a HashMap
+    let left_names = left_on.iter().map(|e| e.name());
```

Comment:
using the `expression.name()` is a bit buggy since it doesn't work on expressions where the output name depends on the schema. we should instead resolve `left_on` on `left.schema()` and for `right` respectively

Reply:
Oh I wasn't aware of this. Is the assumption that names can depend on the schema a recent one? What is it correct to use `Expression::name()`?